sway/ipc-server.c

```diff
@@ -322,6 +331,203 @@ void ipc_get_pixels(wlc_handle output) {
 	ipc_get_pixel_requests = unhandled;
 }
 
+static int ipc_selection_data_cb(int fd, uint32_t mask, void *data) {
+	assert(data);
+	struct get_clipboard_request *req = (struct get_clipboard_request *)data;
+
+	if (mask & WLC_EVENT_ERROR) {
+		sway_log(L_ERROR, "Selection data fd error");
+		goto cleanup;
+	}
+
+	if (mask & WLC_EVENT_READABLE || true) {
+		static const int step_size = 512;
+		char *data = NULL;
+		int ret = 0;
+		int current = 0;
+
+		// read data as long as there is data avilable
+		// grow the buffer step_size in every iteration
+		do {
+			if (data == NULL) {
+				data = malloc(step_size);
+			} else {
+				data = realloc(data, current + step_size);
+			}
+
+			ret = read(fd, data + current, step_size - 1);
+			if (ret < 0) {
+				sway_log_errno(L_ERROR, "Reading from selection data fd failed");
+				goto cleanup;
+			}
+
+			current += ret;
+		} while (ret == step_size - 1);
+
+		data[current] = '\0';
+
+		if (strncmp(req->type, "text/", 5) == 0) {
+			json_object_object_add(req->json, req->type,
+				json_object_new_string(data));
+		} else {
+			size_t outlen;
+			char *b64 = b64_encode(data, current, &outlen);
+			json_object_object_add(req->json, req->type,
+					json_object_new_string(b64));
```

Comment:
Can you add `;base64` to the mimetype? i.e. `image/png` becomes `image/png;base64`